src/components/MerchantList.jsx

```diff
@@ -0,0 +1,105 @@
+import React, { useState, useEffect, useCallback, useRef } from 'react';
+import Merchant from './Merchant';
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+import { faSpinner } from '@fortawesome/free-solid-svg-icons';
+
+const initialMerchants = [
+  { id: 0, name: "海洋大學店", distance: 1.6, costDownLimit: 98, costUpLimit: 123, starRate: 4.8, starNumber: 71 },
+  { id: 1, name: "海", distance: 2.5, costDownLimit: 12, costUpLimit: 124, starRate: 3.2, starNumber: 12 },
+  { id: 2, name: "海洋大學店", distance: 1.6, costDownLimit: 98, costUpLimit: 123, starRate: 4.8, starNumber: 71 },
+  { id: 3, name: "海", distance: 2.5, costDownLimit: 12, costUpLimit: 124, starRate: 3.2, starNumber: 12 },
+  { id: 4, name: "海洋大學店", distance: 1.6, costDownLimit: 98, costUpLimit: 123, starRate: 4.8, starNumber: 71 },
+  { id: 5, name: "海", distance: 2.5, costDownLimit: 12, costUpLimit: 124, starRate: 3.2, starNumber: 12 },
+];
+
+function MerchantList() {
+  const [merchants, setMerchants] = useState(initialMerchants);
+  const [hasMore, setHasMore] = useState(true);
+  const [loading, setLoading] = useState(false);
+
+  const observer = useRef(); // 保存 IntersectionObserver 实例
+  const lastElementRef = useRef(null); // 用于跟踪最后一个 Merchant 元素
+
+  const loadMoreMerchants = useCallback(() => {
```

Comment:
This shouldn't use `useCallback`, because the function will fetch something. So `useCallback` won't have performance accelerating.
So you need to pick the code that is not easy to change the function state, and new a `useCallbak` function.
for example `IntersectionObserver` part